File: src/main/java/RemoveDuplicatesfromSortedList/BIGBOSS/RemoveDuplicatesfromSortedList.cpp
//
// Created by BIGBOSS on 2019-06-27.
//

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution
{
    public:

    ListNode* deleteDuplicates(ListNode* head)
    {
        if (!head)
            return nullptr;

        ListNode* result = head;
        ListNode* cur = head;

        head = head->next;

        while (head)
        {
            if (head->val != cur->val)
            {
                cur->next = head;
                cur = head;
            }

            head = head->next;
        }

        cur->next = nullptr;

        return result;
    }
};

File: src/main/java/RemoveDuplicatesFromSortedList/kunlingou/RemoveDuplicatesFromSortedList.java
package RemoveDuplicatesFromSortedList.kunlingou;

public class RemoveDuplicatesFromSortedList {
	public static void main(String[] args) {
		ListNode h = new ListNode(1);
		ListNode c;
		h.next = new ListNode(1);
		c = h.next;
		c.next = new ListNode(1);
		deleteDuplicates(h);
		System.out.println();
	}
	
	public static ListNode deleteDuplicates(ListNode head) {
		if(head == null) return head;
        ListNode cur = head;
        while(cur.next != null){
            if(cur.val == cur.next.val)
                cur.next = cur.next.next;
            else
                cur = cur.next;
        }
        return head;
    }
}
class ListNode {
     int val;
     ListNode next;
     ListNode(int x) { val = x; }
}